http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java

```diff
@@ -94,6 +96,9 @@ public class HttpServerConfig
 
     private boolean compressionEnabled = true;
 
+    private DataSize maxHeapMemory = DataSize.of(1, GIGABYTE);
+    private DataSize maxDirectMemory = DataSize.of(256, MEGABYTE);
```

Comment:
What's the size in Jetty's default configuration?

Reply:
Which seems like a lot and we didn't account for that in our heap headroom estimatons

Reply:
```
    private long maxMemory(long maxMemory)
    {
        if (maxMemory < 0)
            return -1;
        if (maxMemory == 0)
            return Runtime.getRuntime().maxMemory() / 8;
        return maxMemory;
    }
```

Reply:
My concern is that it may be too large for applications that run with much less memory. I would suggest we make the default 1/8th of the heap with a cap at that value, unless overridden via config with a specific reservation.

Reply:
Also, what's the corresponding default for "maxHeapMemory"?

Reply:
both heap and direct are set to 1/8th of the max memory

Reply:
Done